src/angular/components/register.component.ts

```diff
@@ -1,4 +1,5 @@
 import { Router } from '@angular/router';
+import * as crypto from 'crypto';
```

Comment:
this breaks in web clients like browser extension. node is not available

Reply:
What would be an alternative?

I had trouble following the Typescript. If there is a module with HMAC abilities somewhere in the library where would it be?

Reply:
You would need to use an abstraction of crypto functions, which currently isn't available (will be pulled out of the crypto service where they are currently). It's something I started working on since it will be needed for our CLI and we can also switch the desktop app to.

Reply:
https://stackoverflow.com/a/47332317

Found this... hmmm

Reply:
Yes, there are two WIP cryptoFunction services: https://github.com/bitwarden/jslib/blob/master/src/services/nodeCryptoFunction.service.ts and https://github.com/bitwarden/jslib/blob/master/src/services/webCryptoFunction.service.ts